From 73662d021d4af3e29e7c8c6e85c31a74c1a584ef Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Tue, 19 Oct 2010 09:43:30 +0800
Subject: [PATCH] ENGR00132758 correct NFC_CLK definition

This patch is used to fix the issue caused by ENGR00132709.
NFC_CLK definition should be used in cmd_clk interface.
MXC_NFC_CLK should be used as internal clock name.

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 board/freescale/mx53_rd/mx53_rd.c |    2 +-
 common/cmd_clk.c                  |    4 ++--
 cpu/arm_cortexa8/mx53/generic.c   |    5 ++++-
 include/asm-arm/clock.h           |    3 ++-
 4 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/board/freescale/mx53_rd/mx53_rd.c b/board/freescale/mx53_rd/mx53_rd.c
index 9c71514..596dfd7 100644
--- a/board/freescale/mx53_rd/mx53_rd.c
+++ b/board/freescale/mx53_rd/mx53_rd.c
@@ -1041,7 +1041,7 @@ void setup_nfc(void)
 	 * limit NFC clocks as 34MHZ. The user can modify
 	 * it according to dedicate NAND flash
 	 */
-	clk_config(0, 34, MXC_NFC_CLK);
+	clk_config(0, 34, NFC_CLK);
 
 #if defined(CONFIG_MX53_ARD)
 	mxc_request_iomux(MX53_PIN_NANDF_CS0,
diff --git a/common/cmd_clk.c b/common/cmd_clk.c
index b4130f3..d6f0552 100644
--- a/common/cmd_clk.c
+++ b/common/cmd_clk.c
@@ -43,7 +43,7 @@ int do_clkops(cmd_tbl_t *cmdtp, int flag, int argc, char *argv[])
 		else if (strcmp(argv[1], "ddr") == 0)
 			clk_info(DDR_CLK);
 		else if (strcmp(argv[1], "nfc") == 0)
-			clk_info(MXC_NFC_CLK);
+			clk_info(NFC_CLK);
 		else
 			printf("Unsupported clock type!\n");
 		break;
@@ -56,7 +56,7 @@ int do_clkops(cmd_tbl_t *cmdtp, int flag, int argc, char *argv[])
 		else if (strcmp(argv[1], "ddr") == 0)
 			clk_config(CONFIG_REF_CLK_FREQ, freq, DDR_CLK);
 		else if (strcmp(argv[1], "nfc") == 0)
-			clk_config(CONFIG_REF_CLK_FREQ, freq, MXC_NFC_CLK);
+			clk_config(CONFIG_REF_CLK_FREQ, freq, NFC_CLK);
 		else
 			printf("Unsupported clock type!\n");
 		break;
diff --git a/cpu/arm_cortexa8/mx53/generic.c b/cpu/arm_cortexa8/mx53/generic.c
index 6ba61ee..6e57d9e 100644
--- a/cpu/arm_cortexa8/mx53/generic.c
+++ b/cpu/arm_cortexa8/mx53/generic.c
@@ -643,6 +643,9 @@ int clk_info(u32 clk_type)
 		printf("DDR Clock: %dHz\n",
 			mxc_get_clock(MXC_DDR_CLK));
 		break;
+	case NFC_CLK:
+		printf("NFC Clock: %dHz\n",
+			 mxc_get_clock(MXC_NFC_CLK));
 	case ALL_CLK:
 		printf("cpu clock: %dMHz\n",
 			mxc_get_clock(MXC_ARM_CLK) / SZ_DEC_1M);
@@ -967,7 +970,7 @@ int clk_config(u32 ref, u32 freq, u32 clk_type)
 		if (config_ddr_clk(freq))
 			return -1;
 		break;
-	case MXC_NFC_CLK:
+	case NFC_CLK:
 		if (config_nfc_clk(freq))
 			return -1;
 		break;
diff --git a/include/asm-arm/clock.h b/include/asm-arm/clock.h
index 41f8c8c..fe2813e 100644
--- a/include/asm-arm/clock.h
+++ b/include/asm-arm/clock.h
@@ -34,7 +34,8 @@ enum {
 	UART_CLK,
 	CSPI_CLK,
 	DDR_CLK,
-	ALL_CLK
+	NFC_CLK,
+	ALL_CLK,
 };
 
 int clk_config(u32 ref, u32 freq, u32 clk_type);
-- 
1.5.4.4

